### ISSUE TITLE
Mismatched variant nesting generates [object Object] in class

### ISSUE THREAD

**Issue by Cael6** [nextui-org/tailwind-variants#165]
**Describe the bug**
When using the extend capabilities, if a variation's value does not match the shape of the extended tv result, it will add `[object Object]` to the class list rather than calculate the value.

For example:
```

const styles = tv({
  slots: {
    base: "button",
    other: "other",
  },
  variants: {
    variation: {
      blue: {
        base: "blue",
        other: "other-blue",
      },
    },
  },
});

const extendedStyles = tv({
  extend: styles,
  variants: {
    variation: {
      blue: "extended-blue",
    },
  },
});
```

`styles` uses an object to apply variations to `variation`, but `extendedStyles` uses a string to apply to just the `base`.

**To Reproduce**

1. Go to [codesandbox example](https://codesandbox.io/p/devbox/white-wildflower-forked-rt357f?file=%2Fsrc%2FApp.tsx%3A38%2C29&workspaceId=14b0e31b-cfb1-4cf5-a8ad-89c4d76bcd3b&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsxpg6y900073b6gmcmt5icf%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsxpg6y900043b6gcvkw9b02%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsxpg6y900053b6gnbkntnoc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsxpg6y900063b6gyh10sc7w%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B43.99137657025193%252C56.00862342974807%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsxpg6y900043b6gcvkw9b02%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsxpg6y800033b6g12lmqhg7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A38%252C%2522startColumn%2522%253A29%252C%2522endLineNumber%2522%253A38%252C%2522endColumn%2522%253A29%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsxpg6y900043b6gcvkw9b02%2522%252C%2522activeTabId%2522%253A%2522clsxpg6y800033b6g12lmqhg7%2522%257D%252C%2522clsxpg6y900063b6gyh10sc7w%2522%253A%257B%2522id%2522%253A%2522clsxpg6y900063b6gyh10sc7w%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clsxphasn00663b6gxksssdho%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clsxphasn00663b6gxksssdho%2522%257D%252C%2522clsxpg6y900053b6gnbkntnoc%2522%253A%257B%2522id%2522%253A%2522clsxpg6y900053b6gnbkntnoc%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clsxph6zw004w3b6go8jszqer%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clsxph6zw004w3b6go8jszqer%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A24.270833333333343%257D)
2. Inspect Button's `class`

**Expected behavior**
I expect that both `blue` and `extended-blue` are applied to Button's `class`.

**Screenshots**
![image](https://github.com/nextui-org/tailwind-variants/assets/2346697/ebb56396-6786-43b3-a4bd-504c27764f86)

**Desktop (please complete the following information):**

- OS: macOS
- Browser: Chrome
- Version: 121.0.6167.184 (Official Build) (arm64)

**Additional context**
I'm using TypeScript 5.3.3, but on previous versions of TypeScript I believe it would simply omit `[object Object]` instead of adding in the desired class.

**Comment by thefalked** [comment]
For me it would make sense to use like this:

```js
const styles = tv({
  slots: {
    base: "button",
    other: "other",
  },
  variants: {
    variation: {
      blue: {
        base: "blue",
        other: "other-blue",
      },
    },
  },
});

const extendedStyles = tv({
  extend: styles,
  variants: {
    variation: {
      blue: {
        base: "extended-blue",
      },
    },
  },
});
```

or if you want to add the `blue: "extended-blue"` it would apply to all slots, not just the base one.

keep in mind that `base` is the name of the slot not the "base" for all slots.

**Comment by Cael6** [comment]
Yes, it would work if you matched the structure between `styles` and `extendedStyles`, but it should also work with my example as well. 

Consider a case where `styles` is defined in a library and `extendedStyles` is defined in an application using that library. The types allow you to define `extendedStyles` as I did in the example, so it ends up **not working** as expected because the original `blue` class from `styles` does not get applied.

It is a very difficult bug to track down if you're not aware of the problem.
